crates/aptos/src/move_tool/mod.rs

```diff
@@ -1101,6 +1105,58 @@ impl CliCommand<TransactionSummary> for RunFunction {
     }
 }
 
+/// Run a Move function
+#[derive(Parser)]
+pub struct ViewFunction {
+    /// Function name as `<ADDRESS>::<MODULE_ID>::<FUNCTION_NAME>`
+    ///
+    /// Example: `0x842ed41fad9640a2ad08fdd7d3e4f7f505319aac7d67e1c0dd6a7cce8732c7e3::message::set_message`
+    #[clap(long)]
+    pub(crate) function_id: MemberId,
+
+    /// Arguments combined with their type separated by spaces.
+    ///
+    /// Supported types [u8, u16, u32, u64, u128, u256, bool, hex, string, address, raw, vector<inner_type>]
+    ///
+    /// Example: `address:0x1 bool:true u8:0 u256:1234 'vector<u32>:a,b,c,d'`
```

Comment:
we really need to get to using the same api that rest does

Reply:
You're saying just taking in JSON strings?  We could do that.  But when it uses BCS, it's harder to do from a command line.

Or are you saying automatic BCS conversion